src/app/network/DownloadManager.ts

```diff
@@ -164,79 +164,106 @@ export class DownloadManager {
     }
   };
 
+  private static readonly handleConnectionLostError = (err: unknown, taskId: string) => {
+    tasksService.updateTask({
+      taskId,
+      merge: { status: TaskStatus.Error, subtitle: t('error.connectionLostError') as string },
+    });
+    throw err;
+  };
+
+  private static readonly handleServerOrCancelledError = (
+    downloadTask: DownloadTask,
+    isServerError: boolean,
+    filePickerCancelled: boolean,
+  ): boolean => {
+    if (!isServerError && !filePickerCancelled) return true;
+
+    const retryTasks = RetryManager.getTasksById(downloadTask.taskId);
+    if (retryTasks.length > 0) {
+      tasksService.updateTask({
+        taskId: downloadTask.taskId,
+        merge: { status: TaskStatus.Success },
+      });
+      return false;
+    }
+
+    this.removeRetryItems(downloadTask.items, downloadTask);
+    return true;
+  };
+
+  private static readonly reportErrorWithContext = (err: unknown, items: DownloadItemType[]) => {
```

Comment:
Does this function work for any error you want to report? Or is it for a specific case?